share_app/src/main/java/org/odk/share/activities/WifiActivity.java

```diff
@@ -307,4 +301,89 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
         return super.onOptionsItemSelected(item);
     }
+
+    @Override
+    protected Dialog onCreateDialog(int id) {
+        switch (id) {
+            case DIALOG_DOWNLOAD_PROGRESS:
+                progressDialog = new ProgressDialog(this);
+                progressDialog.setMessage("Receiving");
+                progressDialog.setIndeterminate(true);
+                progressDialog.setCancelable(false);
+
+                progressDialog.setButton(DialogInterface.BUTTON_NEGATIVE, getString(R.string.cancel),
+                        new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int which) {
+                                if (wifiReceiveTask != null) {
+                                    wifiReceiveTask.cancel(true);
+                                }
+                                dialog.dismiss();
+                                finish();
+                            }
+                        });
+                progressDialog.show();
+                return progressDialog;
+            case DIALOG_CONNECTING:
+                progressDialog = new ProgressDialog(this);
+                progressDialog.setMessage("Connecting to wifi");
+                progressDialog.setIndeterminate(true);
+                progressDialog.setCancelable(false);
+                progressDialog.show();
+                return progressDialog;
+            default:
+                return null;
+        }
+    }
+
+    public static String getAccessPointIpAddress(Context context) {
```

Comment:
This method can be moved to WifiHelper since it is a helper method which is not coupled with it's usage